### DISCUSSION TITLE
Not working with UIkit uk-modal

### DISCUSSION THREAD

**Discussion by hkalexling** [alpinejs/alpine#1159] [1. Help] [answered]
Hi, 

I have been using Alpine with [UIkit](https://getuikit.com/docs) and everything works great, except that the `uk-modal` components are not being reactive. Here's a codepen showing the issue: https://codepen.io/hkalexling/pen/GRNBjZE

So we have two buttons (A and B), and when either of them is clicked, we set an alpine property `text` to either `A` or `B`, depending on the button being pressed, and display a `uk-modal`. The value of `text` should be displayed in the modal using `x-text`.

#### Expected behavior:

When button A is pressed, the modal is shown, and the text in the modal is `A`.

#### Actual behavior:

When buttin A is pressed, the modal is shown, but the text in the modal is the initial value `original`.

Am I missing something here?

**Comment by ryangjchandler** [comment, marked as answer]
@hkalexling When UIKit sets up the modal and adds all of it's own JavaScript, it moves it outside of the Alpine component. 

If you inspect, you'll see that it's near the end of the body. This means that when you update the state on your Alpine component and then Alpine tries to update all of the elements, it won't be able to update the modal because it's no longer **inside** of the component.

I've never used UIKit so not sure if you can prevent this behaviour or not.

**Reply by hkalexling** [reply]
Ah right! I did inspect but didn't realize it's outside of the component 🤦 

Thanks a lot for your quick reply! I will investigate and see what I can do.

**Comment by hkalexling** [comment]
In case anyone else's is having the same issue, you can simply do `uk-modal="container: false"` to force the modal to remain in place. See https://github.com/uikit/uikit/issues/3199
